### ISSUE TITLE
Extra 'Expression.Compile()' overload aligned with 'DynamicMethod' options

### ISSUE THREAD

**Issue by vermorel** [dotnet/corefx#20430]
The [extended constructor of DynamicMethod](https://msdn.microsoft.com/en-us/library/0d9fd769(v=vs.110).aspx) offers two important options:

* the target module
* a `skipVisibility` flag

that are very convoluted to replicate from from the [LamdbaExpression.CompileToMethod() method](https://msdn.microsoft.com/en-us/library/system.linq.expressions.lambdaexpression.compiletomethod(v=vs.110).aspx) angle. It would be very practical to provide an extra overload to `CompileToMethod()` with arguments aligned with the options available to the `DynamicMethod` constructor.

**Comment by svick** [comment]
`LamdbaExpression.CompileToMethod()` does not create a new method, it provides the IL for an existing `MethodBuilder`. I think to achieve what you want while keeping the interface consistent, an overload of `CompileToMethod()` that takes `DynamicMethod` instead of `MethodBuilder` could be added.

**Comment by vermorel** [comment]
Yes, agreed, the overload `LamdbaExpression.CompileToMethod(DynamicMethod)` would be superior indeed.
